### PR TITLE
python3Packages.pylama: 7.7.1 -> 8.0.6

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
https://github.com/klen/pylama/blob/8.0.6/Changelog

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pydocstyle/default.nix [hunk 1]
```diff
@@ -1,33 +1,40 @@
...
+    pytestCheckHook
+  ];
 
-  checkPhase = ''
-    # test_integration.py installs packages via pip
-    py.test --cache-clear -vv src/tests -k "not test_integration"
-  '';
+  disabledTestPaths = [
+    "src/tests/test_integration.py"
+  ];
```
Comment (samuela) [inline pkgs/development/python-modules/pydocstyle/default.nix:33]: Please leave a comment explaining why this test path is disabled.

File: pkgs/development/python-modules/pylama/default.nix [hunk 2]
```diff
@@ -1,34 +1,66 @@
...
+    pytestCheckHook
+  ];
 
-  # tries to mess with the file system
-  doCheck = false;
+  disabledTests = [
+    "test_pylint" # infinite recursion
+    "test_quotes" # FIXME package pylama-quotes
+    "test_radon" # FIXME package radon
+  ];
```
Comment (samuela) [inline pkgs/development/python-modules/pylama/default.nix:54]: These comments aren't clear to me. Why is `test_pylint` failing with an infinite recursion? Is this an issue upstream? If so, we should link to it.

What are these FIXMEs for exactly?

**Review by samuela — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pylama/default.nix (hunk 2 above)
Comment (dotlambda) [reply pkgs/development/python-modules/pylama/default.nix:54]: pylint -> isort -> pylama

The FIXMEs are extra requires.

**Comment by dotlambda** [comment]
I removed the pylint dependency from trio because that makes cryptography depend on it and is probably the only reason this causes so many rebuilds.

File: pkgs/development/python-modules/pylama/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/pylama/default.nix:54]: We could move the tests to passthru.tests to break the infinite recursion. It is always good to test package which cause mass rebuilds indepth.

Comment (dotlambda) [reply pkgs/development/python-modules/pylama/default.nix:54]: We are still testing the package, just not the pylint component.
Also after this PR way fewer packages should depend on pylama.

File: pkgs/development/python-modules/pydocstyle/default.nix [hunk 3]
```diff
@@ -1,33 +1,40 @@
...
   disabled = !isPy3k;
 
+  format = "setuptools";
+
   src = fetchFromGitHub {
     owner = "PyCQA";
     repo = pname;
     rev = version;
-    sha256 = "03z8miyppm2xncrc9yjilwl7z5c5cpv51zha580v64p8sb2l0j7j";
+    hash = "sha256-j0WMD2qKDdMaKG2FxrrM/O7zX4waJ1afaRPRv70djkE=";
```
Comment (jonringer) [inline pkgs/development/python-modules/pydocstyle/default.nix:20]: I would prefer it to be sha256, I think a lot of the update scripts look for it

**Review by jonringer — COMMENTED** [review]
otherwise lgtm

Everything else can be done in staging-next

File: pkgs/development/python-modules/pydocstyle/default.nix (hunk 3 above)
Comment (jonringer) [reply pkgs/development/python-modules/pydocstyle/default.nix:20]: confirmed, it does look for the `sha256` string

Comment (dotlambda) [reply pkgs/development/python-modules/pydocstyle/default.nix]: changed it

Comment (mweinelt) [reply pkgs/development/python-modules/pydocstyle/default.nix]: You mean `nix-update`? Works fine for me with `hash` instead of `sha256`.

Comment (jonringer) [reply pkgs/development/python-modules/pydocstyle/default.nix]: I meant `./maintainers/scripts/update-python-libraries`

Comment (SuperSandro2000) [reply pkgs/development/python-modules/pydocstyle/default.nix]: Turns out I was right that not yet everything supports hash.

Comment (mweinelt) [reply pkgs/development/python-modules/pydocstyle/default.nix]: And we can fix that. What a time to be alive!

fwiw: fetchpatch also lacks support for `hash`.

**Pull request merged**